lab/13-npm/classes.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+class Person {
+    constructor(cnp){
+        this.cnp = cnp;
+    }
+};
+
+class Customer extends Person {
+    constructor (cnp, money){
+        super(cnp);
+        this.money = money;
+        this.shoppingCart = [];
+    }
+
+    addToCart(product, quantity, available_products){
+        var available = false;
+        var today = new Date();
+        //available_products is an array in which the product is looked for
+        //(a really small database).
+        //This way of implementing gives a more natural approach when calling
+        //the addToCart method.
+        for (let i = 0; i < available_products.length; i++){
+                //if the wanted product is found in the array
+                if (product === available_products[i].name){
+                    available = true;
+                    //checking the expiration date
+                    if (available_products[i].expirationDate < today)
+                        throw 'Product is expired!';
```

Comment:
You should throw instances of Error (or subclasses of Error), otherwise the stack trace will not be recorded (so you will not know the chain of function calls that lead to that exception).